cli/Valet/Brew.php

```diff
@@ -289,15 +280,48 @@ public function linkedPhp()
 
         return $this->supportedPhpVersions()->first(
             function ($version) use ($resolvedPhpVersion) {
-                $resolvedVersionNormalized = preg_replace('/[^\d]/', '', $resolvedPhpVersion);
-                $versionNormalized = preg_replace('/[^\d]/', '', $version);
-
-                return $resolvedVersionNormalized === $versionNormalized;
+                return $this->isPhpVersionsEqual($resolvedPhpVersion, $version);
             }, function () use ($resolvedPhpVersion) {
                 throw new DomainException("Unable to determine linked PHP when parsing '$resolvedPhpVersion'");
             });
     }
 
+    /**
+     * Extract PHP executable path from PHP Version.
+     *
+     * @param  string  $phpVersion
```

Comment:
Could we specify this should be a `php@8.1` style, since sometimes we're only passing around `8.1`?

```suggestion
     * @param  string  $phpVersion  For example, "php@8.1"
```